packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -214,7 +229,10 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
           // there will be an announcement to "click to finish selecting range" (above).
           state.selectDate(date);
           let nextDay = date.add({days: 1});
-          if (isSameMonth(date, nextDay)) {
+          if (state.isInvalid(nextDay)) {
+            nextDay = date.subtract({days: 1});
+          }
+          if (!state.isInvalid(nextDay)) {
```

Comment:
I don't think we need this check, we cannot possibly be on an invalid day at this point?

Reply:
You could be if you had a where the date before and after are both unavailable. Then we could not advance in either direction.